migration/README.md

```diff
@@ -0,0 +1,124 @@
+        3b. Child chain execution
+
+## Pre-requisites
+
+Before starting the migration process, it is important to gather some information that will be necessary for the coming steps.
+
+The following information is necessary for the migration: 
+
+    - Rollup proxy addres: The address of proxy contract managing the rollup logic for your orbit chain.
+
+    - Parent chain id: The chain id of the parent chain that has your rollup contracts
+
+    - Child chain id: The chain id for your orbit chain
+    
+    - Parent chain RPC URL: The RPC URL for the parent chain that has your rollup contracts 
+    
+    - Child chain RPC URL: The RPC URL for your orbit chain
+
+    - Hotshot address: The address of the hotshot client contract on the parent chain.
+
+    - Parent chain upgrade executor address: The address of the upgrade executor for your rollup on the parent chain
+
+    - Child chain upgrade executor address: The address of the upgrade executor for your rollup on the child chain.
+
+    - New wasm module root: The new wasmModuleRoot of the STF function from the espresso integration. This is critical to enable fraud proofs post migration.
+
+    - Current wasm module root: The current wasm module root that your chain has. This is critical for maintaining backwards compatible fraud proofs for blocks before the migration.
+
+    - Current osp entry: The address of the current osp entry on the parent chain. This is critical to maintain backwards compatible fraud proofs.
+
+    - Proxy admin: The address of the proxy admin for your rollup on the parent chain
+
+    - Upgrade timestamp: The unix timestamp you wish to perform the ArbOS upgrade at.
+
+We would reccommend that you aggregate these in a .env file in variables of the same names to the ones in .example-env so that you can export them when running the commands presented in future sections of this guide.
+
+#### A note on Upgrade Executors
+
+This migration presumes there to be upgrade executor contracts that are chain/rollup owners on both the parent chain, as well as the child chain. If you do not have an upgrade executor on the child chain, the contract execution commands at the end of this guide will not work.
+
+## Contract Deployments
+
+### Parent chain deployments
+
+The parent chain is where most of the contract deployments will happen for the migration to espresso. There are two forge scripts that you need to deploy on the parent chain. These scripts are located in the following directories:
+
+```
+orbit-actions/contracts/parent-chain/espresso-migration/DeployEspressoOsp.s.sol
+```
+and
+```
+orbit-action/contracts/parent-chain/espresso-migration/DeployEspressoOspMigrationAction.s.sol
+```
+
+These must ***MUST*** be deployed in the following order:
+
+1. DeployEspressoOsp.s.sol
+2. DeployEspressoOspMigrationAction.s.sol
+
+In between these deployments you need to record the address of the newly deployed OneStepProverEntry contract and export it in the env var `NEW_OSP_ENTRY` 
```

Comment:
Is this obvious from the output? Or should we add a `jq` command here?

Reply:
I think the output of that command would be obvious if you're running the same verbosity level on the forge script. But I think it also makes sense to add the command anyways. as we know exactly how forge will output the broadcast file.